backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java

```diff
@@ -1695,6 +1699,218 @@ void testSearchWithType() {
+
+        // Create the properties for the endpoint. Leave the URL so that we can set it afterwards.
+        final var camelProperties = new CamelProperties();
+        camelProperties.setBasicAuthentication(new BasicAuthentication(username, password));
+        camelProperties.setDisableSslVerification(disableSslVerification);
+        camelProperties.setSecretToken(secretToken);
+
+        final var webhookProperties = new WebhookProperties();
+        webhookProperties.setBasicAuthentication(new BasicAuthentication(username, password));
+        webhookProperties.setDisableSslVerification(disableSslVerification);
+        webhookProperties.setMethod(method);
+        webhookProperties.setSecretToken(secretToken);
+
+        // Create an endpoint without the type and the properties set.
+        final var name = "endpoint-invalid-urls-name";
+        final var description = "endpoint-invalid-urls-description";
+        final var enabled = true;
+        final var serverErrors = 0;
+        final var subType = "slack";
+
+        final var endpoint = new Endpoint();
+        endpoint.setDescription(description);
+        endpoint.setEnabled(enabled);
+        endpoint.setName(name);
+        endpoint.setServerErrors(serverErrors);
+        endpoint.setSubType(subType);
+
+        // Create a simple class to make testing easier.
+        final class TestCase {
+            public final String expectedErrorMessage;
+            public final String[] testUrls;
+
+            TestCase(final String expectedErrorMessage, final String[] testUrls) {
+                this.expectedErrorMessage = expectedErrorMessage;
+                this.testUrls = testUrls;
+            }
+        }
+
+        // Create all the test cases we will be testing in this test.
+        final List<TestCase> testCases = new ArrayList<>();
+        testCases.add(
+            new TestCase(ValidNonPrivateUrlValidator.INVALID_URL, ValidNonPrivateUrlValidatorTest.malformedUrls)
+        );
+
+        testCases.add(
+            new TestCase(ValidNonPrivateUrlValidator.INVALID_URL, ValidNonPrivateUrlValidatorTest.malformedUris)
+        );
+
+        testCases.add(
+            new TestCase(ValidNonPrivateUrlValidator.INVALID_SCHEME, ValidNonPrivateUrlValidatorTest.invalidSchemes)
+        );
+
+        testCases.add(
+            new TestCase(ValidNonPrivateUrlValidator.PRIVATE_IP, ValidNonPrivateUrlValidatorTest.internalHosts)
+        );
+
+        testCases.add(
+            new TestCase(
+                ValidNonPrivateUrlValidator.UNKNOWN_HOST,
+                ValidNonPrivateUrlValidatorTest.unknownHosts
+            )
+        );
+
+        // Test the URLs with both camel and webhook endpoints.
+        for (final var testCase : testCases) {
+            for (final var url : testCase.testUrls) {
+                // Test with a camel endpoint.
+                camelProperties.setUrl(url);
+                endpoint.setType(EndpointType.CAMEL);
```

Comment:
The "slack" subtype should be set here instead of above (it doesn't make sense for `webhook` endpoints).

Reply:
Right, I'll make it more accurate. Thanks!